src/Transform/ONNX/ConstProp.cpp

```diff
@@ -503,9 +503,26 @@ class ConstPropSplitPattern : public OpRewritePattern<ONNXSplitOp> {
 
     // Rewrite
     unsigned outputNum = splitOp->getNumResults();
-    int64_t rank = splitOp->input().getType().cast<ShapedType>().getRank();
+    Value splitInput = splitOp->input();
+    int64_t rank = splitInput.getType().cast<ShapedType>().getRank();
     IntegerAttr axisAttr = splitOp->axisAttr();
     ArrayAttr splitAttr = splitOp->splitAttr();
+    if (!splitAttr) {
+      // If split parameter is not specified, it is constructed from output.
+      SmallVector<Attribute, 4> splits;
+      for (int i = 0; i < outputNum; ++i) {
+        Value splitOutput = splitOp->getResults()[i];
+        uint64_t splitAxis = axisAttr.getValue().getSExtValue();
+        RankedTensorType inType =
+            constructRankedTensorType(splitInput.getType().cast<ShapedType>());
```

Comment:
How about using SmallVector constructor instead of a for-loop, like:
```
ArrayRef<int64_t> shape = splitInput.getType().cast<ShapedType>().getShape();
Attribute splitSize =  rewriter.getIntegerAttr(rewriter.getIntegerType(64), shape[splitAxis] / outputNum));
SmallVector<Attribute, 4> splits(outputNum, splitSize);
```

Reply:
Updated. Thanks!